### PR TITLE
tasty-plutus: Add outputsToInputs (#62)

### PR DESCRIPTION
**Pull request by sergesku**
As a solution to Issue #62, we can provide the function:
```
outputsToInputs :: ContextBuilder p -> [Input]
outputsToInputs (ContextBuilder _ outs _ _ _) =
  toList $ fmap (\(Output t v) -> Input t v) outs
```
Usage examples:
```
nextInputs <- outputsToInputs validContext <$ shouldValidate "Valid case" validData validContext
do something with nextInputs
```
```
shouldValidateProducing "Valid case" validData validContext
let nextInputs = outputsToInputs validContext
do something with nextInputs
```
```
nextInputs <- liftA2 (<$) outputsToInputs (shouldValidate "Valid case" validData) validContext
do something with nextInputs
```

### CODE REVIEW

File: tasty-plutus/CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,10 @@ This format is based on [Keep A Changelog](https://keepachangelog.com/en/1.0.0).
 
 ## Unreleased
 
+## 4.1.1 -- 2021-11-26
```
Comment (kozross) [inline tasty-plutus/CHANGELOG.md:7]: This should be 4.2 - we use only two version numbers.

File: tasty-plutus/src/Test/Tasty/Plutus/Internal/Context.hs [hunk 2]
```diff
@@ -253,6 +254,21 @@ makeIncompleteContexts ctxs = map (first sconcat) ctxs2
...
+{- All context outputs are converted to a list of inputs:
+ these can be used to build new contexts.
+
+ = Example
+
+ @
+  foldMap input (outputsToInputs initContext) <> restOfContext
+ @
+
+ @since 4.1.1
```
Comment (kozross) [inline tasty-plutus/src/Test/Tasty/Plutus/Internal/Context.hs:266]: Likewise here, this should be 4.2.

**Review by kozross — CHANGES_REQUESTED** [review]

**Comment by blamario** [comment]
The example line

    nextInputs <- outputsToInputs validContext <$ shouldValidateProducing "Valid case" validData validContext

is using the function `shouldValidateProducing` which I understand is not supposed to exist in this PR. I presume that was meant to be `shouldValidate` instead. This line is equivalent to the next example's corrected

    shouldValidate "Valid case" validData validContext
    let nextInputs = outputsToInputs validContext

The problem with this logic is that it only works if `validContext` specifies the exact output UTxOs. In one use case I had in mind for this functionality, it can't do that because one of the outputs is freshly minted. I was assuming that `shouldValidateProducing`  would return the actual UTxOs produced by the script, not those specified by the context. Is that impossible?

**Comment by sergesku** [comment]
I fixed examples in description.

**Comment by sergesku** [comment]
Validation of the script does not create any additional outputs that are not specified in the ScriptContext. Therefore, all we can do is use the context outputs as inputs for the new context.

**Review by kozross — APPROVED** [review]

**Pull request merged**